### PR TITLE
WIP Introduce runs

### PR DESCRIPTION
**Pull request by kerneltime**
Signed-off-by: Ritesh H Shukla <ritesh@oneconcern.com>

### CODE REVIEW

File: pkg/model/run.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	Filesystems []FS `json:"filesystems" yaml:"filesystems"`
+	Database    []DB `json:"database" yaml:"database"`
+}
+
+type Steps struct {
+	Containers []RunStepContainer `json:"containers" yaml:"containers"`
+}
+
+type RunStepContainer struct {
+	ContainerID string `json:"containerid" yaml:"containerid"`
```
Comment (galvare2) [inline pkg/model/run.go]: Will these ContainerID's be generated by the pipelines tool, or by Datamon?

File: pkg/model/run.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+type DB struct {
+	Context  string `json:"context" yaml:"context"`
+	Repo     string `json:"repo" yaml:"repo"`
+	BundleID string `json:"bundleid" yaml:"bundleid"`
+	Label    string `json:"label" yaml:"label"`
+	Type     string `json:"type" yaml:"type"`
+	Port     int    `json:"port" yaml:"port"`
+}
+
+type DataSetIn struct {
```
Comment (galvare2) [inline pkg/model/run.go]: How are DataSetIn and DataSetOut going to be used?

File: pkg/model/run.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	"time"
+)
+
+type Run struct {
+	Type        string      `json:"type" yaml:"type"` // Type of run
+	Name        string      `json:"name" yaml:"name"` // Name for the run
+	ID          string      `json:"id" yaml:"id"`     // Unique id for an instance of a run. Each time the same Type+Name is run it results in a new ID.
+	Timestamp   time.Time   `json:"timestamp" yaml:"timestamp"`
+	Contributor Contributor `json:"contributor" yaml:"contributor"`
+	Steps       []Steps     `json:"steps" yaml:"steps"`
```
Comment (galvare2) [inline pkg/model/run.go]: Steps here are an array, meaning they are ordered linearly (totally ordered). What about runs with complex DAGs that include steps running in parallel that cannot be ordered linearly in this way?

File: pkg/model/run.go (hunk 1 above)
Comment (kerneltime) [reply pkg/model/run.go]: Pipelines

File: pkg/model/run.go (hunk 2 above)
Comment (kerneltime) [reply pkg/model/run.go]: This is used to capture the data after it has been processed, so the pipelines tool would need to pass this these attributes to the datamon side car (once).

File: pkg/model/run.go (hunk 3 above)
Comment (kerneltime) [reply pkg/model/run.go]: Thanks, will include it.

**Pull request closed without merging**